pkg/defaults/catsrcHelpers.go

```diff
@@ -106,8 +114,8 @@ func ensureCatsrcPresent(client wrapper.Client, def olm.CatalogSource, cluster *
 		return nil
 	}
 
-	if AreCatsrcSpecsEqual(&def.Spec, &cluster.Spec) {
-		logrus.Infof("[defaults] CatalogSource %s default and on cluster specs are same", def.Name)
+	if cluster.Annotations[defaultCatsrcAnnotationKey] == defaultCatsrcAnnotationValue && AreCatsrcSpecsEqual(&def.Spec, &cluster.Spec) {
```

Comment:
So it looks like we stomp the "custom default" catalogs when the defaults are enabled again. For consistency, should we update the annotation as well in this situation?

I imagine the current situation looks like this:
1. disable default
2. default deleted
3. create "custom default" without annotation
4. enable default
5. "custom default" has spec stomped, but doesn't get the annotation
6. disable default
7. default is not deleted because it doesn't have the annotation (?)

Reply:
Yes, thanks for catching this. I discarded a separate patch for this issue that did what you're suggesting and forgot to fold that into this one.